pyshelf/cloud/factory.py

```diff
@@ -8,12 +9,14 @@ def __init__(self, config, logger):
         self.logger = logger
 
     def create_storage(self, bucket_name):
-        c = self.config
         # Although bucketName exists in the config provided it is not
         # required and is not used because we want the ability to change
         # buckets when we want.
-        if not c.get("buckets").get(bucket_name):
+        bucket_config = utils.get_bucket_config(self.config, bucket_name)
+        if bucket_config is None:
             self.logger.warning("Access keys for {0} are not in your config.".format(bucket_name))
             raise BucketConfigurationNotFound(bucket_name)
-        return Storage(c["buckets"][bucket_name]["accessKey"], c["buckets"][bucket_name]["secretKey"],
-                       bucket_name, self.logger)
+
+        storage = Storage(bucket_config["accessKey"], bucket_config["secretKey"], bucket_config["name"], self.logger)
```

Comment:
YOU DO NOT NEED TO FIX THIS!  I just wanted to mention that within a function I am ok with making a long variable name shorter if you need to reference it a lot.  The exception to rule is if there is a very long function, however, if I can find where you assign `c` very simply then it is fine.